lib/matplotlib/tests/test_figure.py

```diff
@@ -581,7 +582,8 @@ def test_invalid_layouts():
     pc = ax.pcolormesh(np.random.randn(2, 2))
     fig.colorbar(pc)
     with pytest.raises(RuntimeError, match='Colorbar layout of new layout'):
-        fig.set_layout_engine("tight")
+        with pytest.warns(UserWarning):
+            fig.set_layout_engine("tight")
```

Comment:
I find it odd that it both warns and then errors (the error means it _doesn't_ change, and thus the warning is incorrect)

Consider moving the check up the call stack one level to [`set_layout_engine`](https://github.com/matplotlib/matplotlib/blob/c029f1d45c0fe4b45b2a140652c9cd49fdfc5968/lib/matplotlib/figure.py#L2582) such that only one of the error or warning occur.